source/extensions/access_loggers/common/access_log_base.cc

```diff
@@ -10,15 +11,15 @@ namespace Common {
 void ImplBase::log(const Http::HeaderMap* request_headers, const Http::HeaderMap* response_headers,
                    const Http::HeaderMap* response_trailers,
                    const StreamInfo::StreamInfo& stream_info) {
-  const Http::HeaderMapImpl empty_headers;
+  ConstSingleton<Http::HeaderMapImpl> empty_headers;
```

Comment:
I think it's potentially slightly slower to write this using the ConstSingleton abstraction rather than writing it directly because you are potentially calling get() multiple times in the function body., each of which has to do some kind atomic lookup. If you just write the code directly it will only do the atomic update once. But I'm OK with this.

And I guess this method is faster if all three forms of headers is specified, since then the atomic update will not happen at all :)

But my guess is that in most cases there will be response and request headers but no trailers so ithe lookup will happen once, and it'll be a wash :)